### PR TITLE
Hat graph example

### PR DESCRIPTION
**Pull request by anushkrishnav**
## PR Summary
With reference to this issue : hat graphs need an example... #17537
I have added the example @timhoffm 
## PR Checklist

- [ ] Has Pytest style unit tests
- [x] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/next_api_changes/* if API changed in a backward-incompatible way

### CODE REVIEW

File: examples/lines_bars_and_markers/hat_graph.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
+"""
+===============================
+Hat Graph with labels
+===============================
+This example shows a how to create a hat graph and how to annotate with labels.
+"""
+import matplotlib
+import numpy as np
+import matplotlib.pyplot as plt
+# initialise labels and a numpy array make sure you have N labels of N number of values in the array
```
Comment (jklymak) [inline examples/lines_bars_and_markers/hat_graph.py:10]: Please fix the flake8 errors...

File: examples/lines_bars_and_markers/hat_graph.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        height=int(heights[i])
+        i+=1
+        ax.annotate('{}'.format(height),
+                    xy=(rect.get_x() + rect.get_width() / 2, height),
+                    xytext=(0, 4),  # 4 points vertical offset.
+                    textcoords="offset points",
+                    ha='center', va='bottom')
+Label(playerA,rects1)
+Label(playerB,rects2)
+fig.tight_layout()
```
Comment (jklymak) [inline examples/lines_bars_and_markers/hat_graph.py:45]: Suggest using constrained_layout instead...

File: examples/lines_bars_and_markers/hat_graph.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+playerB = np.array([25, 32, 34, 30, 27])
+x = np.arange(len(labels))
+width = 0.35
+
+fig, ax = plt.subplots()
+rects1 = ax.bar(x - width/2, np.zeros_like(playerA), width,
+                bottom=playerA, label='Player A', fill=False)
+rects2 = ax.bar(x + width/2, playerB - playerA, width,
+                bottom=playerA, label='Player B',edgecolor='black')
+#-----------------------------------------------------------------------------------------
```
Comment (jklymak) [inline examples/lines_bars_and_markers/hat_graph.py:22]: Please remove these.  A carriage return should be adequate to set the block of code off...

File: examples/lines_bars_and_markers/hat_graph.py (hunk 2 above)
Comment (anushkrishnav) [reply examples/lines_bars_and_markers/hat_graph.py:45]: Okay will correct it shortly

File: examples/lines_bars_and_markers/hat_graph.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
+"""
+===============================
+Hat Graph with labels
+===============================
+This example shows a how to create a hat graph and how to annotate with labels.
```
Comment (jklymak) [inline examples/lines_bars_and_markers/hat_graph.py:5]: Maybe add a reference to the article?   
```suggestion
This example shows a how to create a hat graph (https://doi.org/10.1186/s41235-019-0182-3) 
and how to annotate with labels.
```

Comment (anushkrishnav) [reply examples/lines_bars_and_markers/hat_graph.py:5]: Yeh Are we allowed to add references didnt no if it is allowed

**Comment by anushkrishnav** [comment]
@jklymak Thank you so much for your help and time.

**Comment by jklymak** [comment]
Thanks @anushkrishnav.  Congrats on your first matplotlib PR!

**Pull request merged**